### DISCUSSION TITLE
Add `.yarn` to gitignore

### DISCUSSION THREAD

**Discussion by transcental** [expo/expo#18000] [Proposals and Ideas]
### Summary

I would like the `.yarn` folder to be included inside of the default `.gitignore`. Not only should this folder not be committed, it is also extremely hard to remove as it's included in the default commit and is for some reason ignored by my adding it into the `.gitignore`.

### Environment

  expo-env-info 1.0.4 environment info:
    System:
      OS: macOS 10.15.7
      Shell: 5.7.1 - /bin/zsh
    Binaries:
      Node: 16.14.2 - ~/.nvm/versions/node/v16.14.2/bin/node
      Yarn: 3.2.0 - ~/.nvm/versions/node/v17.4.0/bin/yarn
      npm: 8.5.0 - ~/.nvm/versions/node/v16.14.2/bin/npm
      Watchman: 2022.06.20.00 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /Users/dillonbarnes/homebrew/opt/cocoapods/bin/pod
    IDEs:
      Android Studio: Chipmunk 2021.2.1 Patch 1 Chipmunk 2021.2.1 Patch 1
      Xcode: /undefined - /usr/bin/xcodebuild
    npmPackages:
      expo: ~45.0.0 => 45.0.6 
      react: 17.0.2 => 17.0.2 
      react-dom: 17.0.2 => 17.0.2 
      react-native: 0.68.2 => 0.68.2 
      react-native-web: 0.17.7 => 0.17.7 
    npmGlobalPackages:
      expo-cli: 5.4.11
    Expo Workflow: managed

### Reproducible demo

Use `expo init` to create a new project. I used the blank managed workflow template.

Make some changes. Add `.yarn` to the gitignore and commit your changes. `.yarn` is still committed to git.

**Comment by brentvatne** [comment]
this doesn't seem entirely straightforward: https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored

i think it's up to the developer to decide what they want to commit here. any suggestions how we can handle this for yarn 2+ users?

**Reply by transcental** [reply]
From what is on the documentation you sent me, it seems quite straightforward.
I would suggest adding the non-zero-installs version because the only difference is the `.pnp.` file which could easily be removed by the developer.

As for yarn 2+, why would you support older versions? Surely the point of the newer versions is that they're better so therefore you should upgrade from the previous versions? Personally, I use yarn 3.2.0.

**Reply by brentvatne** [reply]
i believe most of the yarn community uses yarn classic, because a primary driver for new yarn versions was to add pnp support - which is not compatible with react-native, and so you need to fall back to the node_modules linker. feel free to send a pr for the gitignore changes, they live in the templates directories: https://github.com/expo/expo/tree/main/templates

**Comment by transcental** [comment]
Can be closed by #18021
